doc/neps/nep-0022-roadmap.rst

```diff
@@ -0,0 +1,208 @@
+
+There was a `discussion <http://numpy-discussion.10968.n7.nabble.com/A-roadmap-for-NumPy-longer-term-planning-td45613.html>`_ on the mailing list. We will revisit this at the upcoming `SciPy conference BOF session <https://scipy2018.scipy.org>`_ 
+
+Work on NumPy can be divided into various tasks, all worthy of devloper time investment. Broadly speaking, we can identify the following tasks
+
+Ongoing Maintenance
+~~~~~~~~~~~~~~~~~~~
+
+- triaging issues and pull requests
+- document both python and C-API (i.e. borrowed or new refs)
+
+  - can we document reference borrowing close to C code (in the spirit of doctests for Python code)
+- code coverage in tests
+- actually deprecate things that are deprecated
+- make internals private
+
+Policies for Ensuring Code Quality
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Is this a separate NEP?
+
+- policy for merging pull requests
+
+  - bugfixes - does it fix the bug
+  - releases have their own rules
+  - at least one reviewer
+  - documentation typos can be merged by the committer
+  - enhancements should go through the mailing list
+
+New Functionality
+~~~~~~~~~~~~~~~~~
+
+- Extensible dtypes
+- Overloads for duck arrays [2]_
+- Many new NEPs in progress
+- `Typing stubs for mypy <https://github.com/numpy/numpy-stubs>`_
+
+  - Basic support in NumPy
+  - Standard way to write/check annotations for N-dimensional arrays in Python (this will need a PEP)
+- **Internal refactorings** of almost-external things like MaskedArray, f2py, numpy.distutils, financial
+- How to handle the **Matrix class**; we want to avoid new usage, reduce external dependency
+- Better **configuration reporting** (use it when reporting bugs)
+- Allow switching numpy.linalg backends
+
+Lessons Learned
+---------------
+
+We may not be able to change any of these Right Now.
+
+- Currently strategy is: either pass to ufunc or convert to ndarray. Makes it hard to use NumPy as high-level API, because it is unknown (without reading the code) whether other classes will survive various operations. See discussion above.
+- 0-dim arrays get converted into NumPy scalars
+
+Infrastructure / Ecosystem
+--------------------------
+
+- `multibuild <https://github.com/matthew-brett/multibuild>`_ wheel builder (Linux/OSX/Windows)
+- Airspeed Speed Velocity for benchmarks (https://pv.github.io/numpy-bench/) or 
+  codespeed (https://speed.python.org)
+- Maintain the websites scipy.org, numpy.org
+- numpydoc
+- pytest
+
+  - To what extent should we use pytest APIs?
+  - pytest is a little magical (e.g., fixtures)
+  - Should define a philosophy. Discussion led to a direction of simple use of pytest, using as little magic as feasable.
+- OpenBLAS
+
+Philosophy / scope
+------------------
+
+What is part of NumPy, what is not?
+
+- In memory, N-dimensional single pointer + strided arrays on CPUs
+
+  - Not specialized hardware
+- Basic functionality for scientific computing
```

Comment:
This rst syntax is a little weird - can't you just write

```
- outer list
  - inner list
- outer list
```
or maybe
```
- outer list
  - inner list

- outer list
```

Reply:
RST must have a blank line before the next identation level, so I can do the last one